using_openshift/templates.adoc

```diff
@@ -9,4 +9,47 @@
 
 toc::[]
 
-== Overview
\ No newline at end of file
+== Overview
+A template describes a set of resources intended to be used together that can be customized and processed to produce a configuration.  Each
+template can define a list of environment variables that can be modified for consumption by containers.
+
+== Parameters
+The list of parameters to be overridden are listed in the *parameters* section of the template or by executing the following command:
+
+****
+`osc process --parameters -f [replaceable]#<filename>#`
+****
+
+Following is an example of the output when listing the parameters for one of the *openshift/origin/examples* template.
+The output identifies several parameters that will be generated using an regex expression generator when the template is processed.
+
+----
+$ osc process --parameters -f examples/sample-app/application-template-dockerbuild.json
+NAME                  DESCRIPTION              GENERATOR           VALUE
+ADMIN_USERNAME        administrator username   expression          admin[A-Z0-9]{3}
+ADMIN_PASSWORD        administrator password   expression          [a-zA-Z0-9]{8}
+MYSQL_ROOT_PASSWORD   database password        expression          [a-zA-Z0-9]{8}
+MYSQL_DATABASE        database name                                root
+----
+
+
+== Generating a Configuration
+A configuration results from processing a template.  Adding the appropriate flags allow the parameters to be overriden from the values defined in the file.
+The configuration can be generated by executing the following command:
+
+****
+`osc process -f [replaceable]#<filename>#`
+****
+
+Continuing the previous example, overriding the *ADMIN_USERNAME* and *MYSQL_DATABASE* produces a *json* string that can be applied to the server.
+The *json* string represents a *Config* with customized environment variables.
+
+----
+$osc process -f examples/sample-app/application-template-dockerbuild.json -v ADMIN_USERNAME=root,MYSQL_DATABASE=admin
```

Comment:
I would have gone for ADMIN_PASSWORD as the second parameter to override, myself.